flutter_cache_manager/lib/src/cache_managers/image_cache_manager.dart

```diff
@@ -72,7 +74,7 @@ mixin ImageCacheManager on BaseCacheManager {
       return originalFile;
     }
 
-    var image = decodeImage(await originalFile.file.readAsBytes())!;
+    var image = await _decodeImage(originalFile.file);
```

Comment:
What do you think of checking this instead of only setting allowUpscaling to false? 
This prevents saving the same file twice on disk.
```suggestion
    var image = await _decodeImage(originalFile.file);
    
    var shouldResize = maxWidth != null ? image.width > maxWidth : false ||
    maxHeight != null ? image.height > maxHeight : false;
    if (!shouldResize) return originalFile;
```

Reply:
Sounds quite reasonable!